### ISSUE TITLE
"WARN: The high priority inbound consensus queue is full!

### ISSUE THREAD

**Issue by gidiguru** [Concordium/concordium-node#20]
Bug Description
  An error seems to occur on Baker nodes that are running Concordium Client v0.4.11 (testnet) on WIndows OS, when stopped 
  and restarted. This happens when the node is out of sync with the consensus nodes on the chain, and it fails to catch up with 
  them. 

  Running a node from scratch works fine. This error seems to happen when there is already node data that isn't flushed during 
  the execution of the concordium-client app. Running with or without "--no-block-state-import" produces the same error.
  
  An example of the series of events that lead to the error is below:

$ concordium-node  --no-block-state-import --listen-node-port
  Checking whether docker is accessible on your PATH
  OK
  Checking whether Docker is running
  OK
  Checking whether we can access docker
  OK
  Validating system credentials
  OK
  Currently your node name is set to gidiguru. Press enter to keep it or enter a new one:
  OK
  Latest version of the concordium client is 0.4.11
  Using your local image
  Do you want to flush the local node database before starting? (y/N): n
  Now going to boot up node with name gidiguru running v0.4.11 (testnet)
  INFO: Consensus layer started
  INFO: RPC server started
  INFO: Starting out of band catch-up
  ERROR: External: Error importing block: OtherError ResultDuplicate
  INFO: Starting the P2P layer
  INFO: Attempting to bootstrap
  INFO: Using bootstrapper 54.77.133.220:8888
  INFO: Using bootstrapper 52.17.42.154:8888
  INFO: Commencing baking
  INFO: Runner: Starting baker thread
  ERROR: [sending to x.x.x.x:8888] Connection refused (os error 111)
  INFO: Not enough peers - sending GetPeers requests
  ERROR: [sending to x.x.x.x:8888] Connection refused (os error 111)
  No peers at all - retrying bootstrapping
  INFO: Sent a direct message to peer containing a catch-up status message
  WARN: Couldn't process a finalization message from peer due to error code InvalidResult
  ERROR: [sending to x.x.x.x:8888] Connection refused (os error 111)
  WARN: The high priority inbound consensus queue is full!
  WARN: The high priority inbound consensus queue is full!
   WARN: The high priority inbound consensus queue is full!

Steps to Reproduce
  Any of the following produces the same result
  concordium-node --no-block-state-import --listen-node-port
  concordium-node  --listen-node-port
  concordium-node --no-block-state-import --listen-grpc-port
  concordium-node  --listen-grpc-port
   
Expected Result
  Once the node is restarted, it catches up with the other nodes in the chain.
    
Actual Result
  It doesn't
 
Versions
  Software Version is Concordium Client v0.4.11
   
OS
  Windows OS running GitBash

Node Logs
   https://drive.google.com/file/d/1gIykGV_sdY9KNJt9u7-2FWpb1MC7JqEC/view?usp=sharing

**Comment by abizjak** [comment]
Thank you for the report.

This is unfortunately a known issue, and it is mainly an issue with the way the docker distribution is packaged and the fact that the directory the node is using is mapped to the host. This means that it s very slow to access due to limitations of WSL/Docker on Windows. This, combined with the fact that during catchup and starting from existing state there is a lot of hard drive activity.

There is no solution to this at the moment other than running the linux node distribution under WSL.

**Comment by pablozsc** [comment]
> 
> 
> Thank you for the report.
> 
> This is unfortunately a known issue, and it is mainly an issue with the way the docker distribution is packaged and the fact that the directory the node is using is mapped to the host. This means that it s very slow to access due to limitations of WSL/Docker on Windows. This, combined with the fact that during catchup and starting from existing state there is a lot of hard drive activity.
> 
> There is no solution to this at the moment other than running the linux node distribution under WSL.

Thank you for clarifying this situation! I really didn't know what to do next and how to help people, except how to advise them to switch to linux )

**Comment by NerdsLogic** [comment]
Per Pablo's request, here is the log file for MouseHouse node at: https://nextcloud.dreamymouse.com/nextcloud/s/EBs8z4L3fY9m2Pz

**Comment by gidiguru** [comment]
Thanks for mentioning the option of running the linux node distribution under WSL. I was able to get the node back in sync with the chain by installing docker directly in the WSL2 Ubuntu distribution instead of using docker desktop for windows by proxy which I think is the main issue.

I copied the block data and credentials from the docker windows location into the linux location and restarted concordium-node. It built on the data stream already downloaded from the initial windows run, so it's great that I didn't have to start the download all over.

I was also able to port my baker credentials to the linux distribution, and once the node caught up with the chain, the baker status was reinitiated.

The one thing I noticed though was that using the "no-block-state-import" flag when running concordium node could lead to a corrupt node dataset for some reason, so now I just allow it download from upstream without using the flag when starting the node. It's just a few seconds longer and can save the pain of starting the chain all over.

It will be good to update the docs or at least point people who are using Windows to run nodes in the right direction to save a lot of future frustration. I will be happy to help out if needed.
